### PR TITLE
Fixing everything in the app

### PR DESCRIPTION
**Pull request by chiragw15**
Fixes issue #873 

Changes:
1. Removed Search
2. Removed Action Bar
3. Introduced FAB
4. Removed Themes
5. Added dialog box for copy and sharing
6. Fixed UI of Login, Signup and Forgot password
7. Removed Memory Leaks
8. Fixed crashes and potential crashes
9. Fixed sending layout view
10. Fixed tests
11. Made app stable. 

.

.

.

.

.

.

.

.

.

.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
```
Comment (betterclever) [inline gradle/wrapper/gradle-wrapper.properties]: Is there any specific reason to use a lower gradle version than currently recommend 3.3? I guess you changed it by mistake. Please see to it.

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.0'
+        classpath 'com.android.tools.build:gradle:2.2.3'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.2'
```
Comment (betterclever) [inline build.gradle]: Here as well. Please use '2.3.0' or newer version unless specifically needed.

**Comment by chiragw15** [comment]
@mariobehling @amitiwary999 @betterclever @chashmeetsingh review please.

**Comment by mariobehling** [comment]
These are a lot of changes and reviews take a long time. As we are not in production yet and we have a backlog, plus this is addressing often small items in different places I am merging this as current state. Thank you!

**Pull request merged**